### PR TITLE
Send field to everyone who connect to websocket

### PR DESCRIPTION
**Pull request by bronvic**
#30

Every user will receive field on connection by websocket

Now it looks like this: 
`{"row_n":2,"col_n":2,"field":{"hexes":[{"x":0,"y":0,"unit":null,"content":null},{"x":0,"y":1,"unit":null,"content":null},{"x":1,"y":0,"unit":null,"content":null},{"x":1,"y":1,"unit":null,"content":null}]}}`

### CODE REVIEW

File: backend/src/game_objects/grid.rs [hunk 1]
```diff
@@ -6,6 +6,27 @@ pub struct Grid {
     pub hexes: Vec<Hex>,
 }
 
+impl Grid {
+    pub fn new(row_n: u32, col_n: u32) -> Grid {
+        let hexes = (0..row_n).fold(vec![], |mut acc, x| {
```
Comment (alex-eg) [inline backend/src/game_objects/grid.rs:11]: Why not just
```rust
    let mut hexes: Vec<Hex> = vec![];
    for i in 0..row_n {
        for j in 0..col_n {
            hexes.push(Hex { x: i, y: j, unit: None, content: None });
        }
    }
```
?

Comment (bronvic) [reply backend/src/game_objects/grid.rs:11]: what are this advantages of functional programming for if you can just use two cycles? 

i tired of golang were for .. in is the only option

obviously, i'll fix this

Comment (alex-eg) [reply backend/src/game_objects/grid.rs:11]: For this case, there's none, both approaches are fully equivalent.
However, for more complex ones, there may be many advantages in functional approach.

Also, if you would like to keep things functional, consider avoiding `mut` variables and mutating operations like `append`. I think, more idiomatic functional variant would be something like this:
```rust
let m : Vec<Hex> = (0..row_n).flat_map(
        move |i| (0..col_n).map(
            move |j| {
                Hex { x: i, y: j, unit: None, content: None }
            }).collect::<Vec<Hex>>()
    ).collect();
```

Comment (alex-eg) [reply backend/src/game_objects/grid.rs:11]: Also, in your approach, `acc` is `mut` needlessly. For the `fold` function, the new acc is its return value, so the correct way would be something like this:
```
let hexes = (0..row_n).fold(
    vec![],
    |acc, i| [acc, (0..col_n).map(
        |j| Hex { x: i, y: j, unit: None, content: None }
    ).collect()].concat());
```

File: backend/src/game_objects/grid.rs [hunk 2]
```diff
@@ -14,6 +31,27 @@ mod test {
...
+        assert_eq!(grid.hexes.len(), (row_n * col_n) as usize);
+
+        let mut hexes = grid.hexes.into_iter();
+        for x in 0..row_n {
+            for y in 0..col_n {
+                let hex = hexes.next().unwrap();
+
+                assert_eq!(hex.x, x);
+                assert_eq!(hex.y, y);
+                assert_eq!(hex.unit.is_none(), true);
```
Comment (bronvic) [inline backend/src/game_objects/grid.rs:49]: just assert without _eq

**Pull request merged**